tower/src/builder/mod.rs

```diff
@@ -172,6 +182,41 @@ impl<L> ServiceBuilder<L> {
         }
     }
 
+    /// Buffer requests
+    pub fn buffer(self, bound: usize) -> ServiceBuilder<Chain<BufferLayer, L>> {
+        self.layer(BufferLayer::new(bound))
+    }
+
+    /// Filter requests
+    pub fn filter<U>(self, predicate: U) -> ServiceBuilder<Chain<FilterLayer<U>, L>> {
```

Comment:
should we have a where clause here to bound what `U` should be to not require the user to go to `FilterLayer::new` to find the bound?

Reply:
We cannot add bounds because we don't know the type of `Request`.